package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -622,6 +693,13 @@ def _single_frame(self):
             return_distances=True,
         )
 
+        if np.size(d_a_indices) == 0:
```

Comment:
Do we need `np.size` here, can't we just do `if not d_a_indices:` and simplify the code?

(I don't really know what kind of data structure `d_a_indices` is so please explain.)